Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -78,25 +78,45 @@ extension ModuleError.InvalidLayoutType: FixableError {
 }
 
 extension Module {
-    /// An error in the organization of an individual module.
+    
+    /// An error in the organization or configuration of an individual module.
     enum Error: Swift.Error {
+        
+        /// The module's name is invalid.
+        case invalidName(path: String, name: String, problem: ModuleNameProblem)
+        enum ModuleNameProblem {
```

Comment:
Right, this is very much a work in progress.  The idea is to overhaul the errors here so that they are specific, and this is definitely something that is going to need to grow over time.  I will be adding comments and trying to make these errors (and their fix-its) as specific and clear as possible, but I see the testing module naming as just one aspect of various restrictions on the module name.